sqlmesh/utils/system.py

```diff
@@ -0,0 +1,13 @@
+import os
+import sys
+
+
+def is_daemon_process() -> bool:
+    """
+    Checks if the current process is a daemon process.
+
+    :return: True if the process is a daemon process, False otherwise.
+    :rtype: bool
+    """
+
+    return not os.isatty(sys.stdout.fileno())
```

Comment:
Can you just do:
```
sys.stdout.isatty()
```
?

Reply:
Hey @izeigerman. I believe there's a risk of getting an AttributeError in some cases when the underlying file descriptor doesn't implement isatty(). Or if the stdout is in a wrapper that doesn't correctly implement isatty(). Anyway, it feels like best practice. It would work most of the time with just sys.stdout.isatty(), though.

Reply:
As for your question about why the change: I am working with the airflow integration and after one of the recent new versions (I was on 0.18.0 I think) I started getting errors surrounding the worker settings. I noticed the code in constants.py had a recent change that was supposed restrict the number of workers to 1 when in an airflow environment, but the check for the process being within a daemon was using the multiprocessing package which would only return a "true" if the daemon were a multiprocessing package daemon. It doesn't appear to be that way with airflow, but rather, it's using a standard system daemon for which there's no built-in check. I hope that helps!